### ISSUE TITLE
Codeblocks not rendering upon Refresh or Initial Load

### ISSUE THREAD

**Issue by eduardo-exists** [timlrx/tailwind-nextjs-starter-blog#269]
Hi There!

I am getting this weird issue in production where prism code blocks across my site render correctly when navigating through the site, however, if you open the page directly or refresh the page, code blocks don't render, and instead, only show something like the screenshot below:

![image](https://user-images.githubusercontent.com/23421760/140949065-60ffd8a5-0d87-4680-94b5-12c81a653452.png)

It is very odd because, for a millisecond, the code will render correctly and then turn into looking like the screenshot above.

This doesn't happen in development, and only in production when deploying to vercel. Any ideas on what could be wrong, or how to fix this? I've been banging my head against the wall and trying a bunch of different things :( 

If you check my production version here, you can see the issued -> https://eduardo-tailwind-next.vercel.app/blog/code-sample

**Comment by eduardo-exists** [comment]
More context: I wrote a renderer for Jupyter Notebooks that uses prism-react-renderer to render the code cells. I was getting the same issue on pages rendered in this way, but I was able to solve it by utilizing useEffect and only conditionally rendering after the first load, which is annoying, but it worked -- however, it is still broken on the markdown rendered pages.

**Comment by timlrx** [comment]
Hmm, that's odd. It seems that there is an extra script being rendered on the client side that modifies the code block. Your point on useEffect suggests something related to that as well. Can you share a link to your repository? Will be easier for me to see what's causing it. Thanks.

**Comment by eduardo-exists** [comment]
Thanks so much for being willing to take a look! :)
The repo is: https://github.com/eduardo-exists/edwtf-tailwind
What is odd to me is that if it were to be the prism-react-renderer, that it would be having a side effect on the mdx rendering too.

The specific notebook rendering happens in this file: https://github.com/eduardo-exists/edwtf-tailwind/blob/master/lib/renderNotebook.js

**Comment by timlrx** [comment]
It seems like you managed to get things fixed? I am not able to replicate the error but also saw that you have been making many changes since yesterday. I think the issue is with the interaction of `prism-react-renderer` and `rehype-prism-plus`. You would not want to mix the two as `rehype-prism-plus` converts the code block to nicely formatted html already and  `prism-react-renderer` does something similar but on the client side. Hence you get a properly formatted code block but that flashes to `[object] [object]...` when the js hydrates and `prism-react-renderer`  does the job.

I see that in your new solution you have created a whole new unified processing pipeline to process the notebooks - pretty neat! Otherwise, I would have suggested to output it to a markdown format with `nbconvert` before passing it in the standard mdx pipeline.

**Comment by eduardo-exists** [comment]
Thank you again for your help and your insight! I was able to get things fixed! After your comment, I actually read your blog post on rehype-prism-plus and figured that I should probably use the unified ecosystem to actually compile the notebooks. It seemed like the way I was doing it was an anti-pattern with server-side rendering. So I started over with the unified pipelines and now all the syntax highlighting is done with rehype-prism-plus 👍🏼.

My only complaint is that I haven't been able to figure out how to mimic how the MDX bundler does things to use es-build. I was able to get to the point of using stuff from xdm to get a esast and use the recma plugin to build the jsx, but then I couldn't figure out to actually get something usable. I think the next step is to use esbuild to compile the jsx (?), couldn't figure out how to do that though.

I ended up just stringifying the html AST and converting it to react components with rehype-react - so now it works pretty well but I just wish I could do it more eloquently :D
